crates/sovereign/sovereign-chain-components/src/sovereign/impls/sovereign_to_cosmos/client/update_client_payload.rs

```diff
@@ -50,25 +50,60 @@ where
         target_height: &RollupHeight,
         client_state: Chain::ClientState,
     ) -> Result<SovereignUpdateClientPayload, Chain::Error> {
-        let tm_trusted_height = Height::new(0, trusted_height.slot_number)
-            .map_err(|e| eyre!("Error creating Tendermint Height: {e}"))?;
-        let tm_target_height = Height::new(0, target_height.slot_number)
-            .map_err(|e| eyre!("Error creating Tendermint Height: {e}"))?;
-        let da_trusted_height = DataChainHeight::new(0, trusted_height.slot_number)
-            .map_err(|e| eyre!("Error creating DA Height: {e}"))?;
-        let da_target_height = DataChainHeight::new(0, target_height.slot_number)
-            .map_err(|e| eyre!("Error creating DA Height: {e}"))?;
+        // DA height is higher than rollup height. This requires adding
+        // the genesis Height to the trusted and target Heights
+        let da_trusted_height = Height::new(
+            client_state
+                .sovereign_params
+                .latest_height
+                .revision_number(),
```

Comment:
this should be DA height revision number.

Reply:
Fixed in 1abc19722024204398ae6ce24d41d6aef7ce36ce